spec/helpers/application_helper_spec.rb

```diff
@@ -364,4 +364,91 @@
       end
     end
   end
+
+  describe "#time_in_zone" do
+    let(:user) { create(:user) }
+    let(:pref) { create(:preference) }
+    let(:time) { Time.rfc3339("1999-12-31T16:00:00Z") }
+    let(:utc) { Time.find_zone("UTC") }
+    let(:zone_tokyo) { Time.find_zone("Asia/Tokyo") }
+
+    context "nothing is explicitly specified, current user specifies Asia/Tokyo" do
+      it "formats time in user-specified timezone" do
+        pending "See https://github.com/otwcode/otwarchive/pull/4270"
+
+        allow(pref).to receive(:time_zone).and_return(zone_tokyo)
+        allow(user).to receive(:preference).and_return(pref)
+        allow(User).to receive(:current_user).and_return(user)
+        result = helper.time_in_zone(time)
+
+        expect(result.html_safe?).to eq(true)
+        expect(strip_tags(result)).to eq("Sat 01 Jan 2000 01:00AM JST")
+      end
+    end
+
+    context "argument specifies UTC" do
+      context "no current user (or logged out)" do
+        it "formats time in UTC" do
+          pending "See https://github.com/otwcode/otwarchive/pull/4270"
+
+          result = helper.time_in_zone(time, utc, nil)
+
+          expect(result.html_safe?).to eq(true)
+          expect(strip_tags(result)).to eq("Fri 31 Dec 1999 04:00PM UTC")
+        end
+      end
+
+      context "user preference specifies UTC" do
+        it "formats time in UTC" do
+          pending "See https://github.com/otwcode/otwarchive/pull/4270"
+
+          allow(pref).to receive(:time_zone).and_return(utc)
+          allow(user).to receive(:preference).and_return(pref)
+          result = helper.time_in_zone(time, utc, user)
+
+          expect(result.html_safe?).to eq(true)
+          expect(strip_tags(result)).to eq("Fri 31 Dec 1999 04:00PM UTC")
+        end
+      end
+
+      context "user preference specifies Asia/Tokyo" do
+        it "appends time in user-specified timezone" do
+          pending "See https://github.com/otwcode/otwarchive/pull/4270"
+
+          allow(pref).to receive(:time_zone).and_return(zone_tokyo)
+          allow(user).to receive(:preference).and_return(pref)
+          result = helper.time_in_zone(time, utc, user)
+
+          expect(result.html_safe?).to eq(true)
+          expect(strip_tags(result)).to eq("Fri 31 Dec 1999 04:00PM UTC (01:00AM JST)")
+        end
+      end
+
+      context "user specifies nothing" do
+        it "formats time in UTC, shows '(set time zone)'" do
+          pending "See https://github.com/otwcode/otwarchive/pull/4270"
+
+          allow(pref).to receive(:time_zone).and_return(nil)
+          allow(user).to receive(:preference).and_return(pref)
+          result = helper.time_in_zone(time, utc, user)
+
+          expect(result.html_safe?).to eq(true)
+          expect(strip_tags(result)).to eq("Fri 31 Dec 1999 04:00PM UTC (set timezone)")
+        end
+      end
+    end
+  end
```

Comment:
I added these tests for ``time_in_zone`` after finding out what I think is a bug of the function incidentally. I think I shouldn't try to fix ``time_in_zone`` here, but I hope (pending) tests are acceptable. A previous PR #2453 had to be reverted for changes in that function. Tests like these would help preventing that.

Reply:
Let's definitely not touch `time_in_zone` for now.

I have mixed feelings about pending tests, because we have a few that end up not being enabled for a long time, but still needs to be updated once in a while (e.g. when we need update Ruby or some test libraries). We do have a copy of the pending tests in your commit history here to revisit if necessary.

Reply:
I still believe that function [needs to be fixed](https://otwarchive.atlassian.net/browse/AO3-4494?focusedCommentId=358854) (as shown by the failures of these tests), but I agree that it should be dealt with separately, not here. If it is reported (or gets reported) somewhere, I'd be interested in working on it.

For now, I have removed them per suggestion.